WeakAurasOptions/OptionsFrames/OptionsFrame.lua

```diff
@@ -897,14 +897,32 @@ function OptionsPrivate.CreateFrame()
   dynamicTextCodesFrame.label = dynamicTextCodesLabel
   dynamicTextCodesFrame:Hide()
 
-  function OptionsPrivate.ToggleTextReplacements(data, show, widget)
-    if show or not dynamicTextCodesFrame:IsShown() then
+  function OptionsPrivate.ToggleTextReplacements(data, widget, event)
+    -- This throttle prevents double toggling as a result from OnEditFocusLost
+    local currentTime = GetTime()
+    if event ~= "OnEditFocusGained"
+    and dynamicTextCodesFrame.lastCaller
+    and dynamicTextCodesFrame.lastCaller.event == "OnEditFocusLost"
+    and currentTime - dynamicTextCodesFrame.lastCaller.time < 0.1
+    and not (event == "OnEnterPressed" and dynamicTextCodesFrame:IsMouseOver())
```

Comment:
I don't understand what this is guarding against.

Reply:
When the frame is opened and then the user clicks the ToggleButton to close the frame, that action would first fire OnEditFocusLost and thus hide the frame first before the toggle button's callback runs and show the frame again. 
The throttle prevents that by checking if the frame was recently hidden from OnEditFocusLost.